source/extensions/omni.isaac.lab_tasks/omni/isaac/lab_tasks/utils/wrappers/torchrl/torchrl_env_wrapper.py

```diff
@@ -0,0 +1,601 @@
+# Copyright (c) 2022-2025, The Isaac Lab Project Developers.
+# All rights reserved.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+from __future__ import annotations
+
+import contextlib
+import numpy as np
+import os
+import time
+import torch
+import warnings
+from collections.abc import Iterator
+from dataclasses import asdict
+from tensordict import TensorDict, TensorDictBase, unravel_key
+from tensordict.nn import ProbabilisticTensorDictSequential, TensorDictModule, dispatch
+from typing import Any
+
+from torchrl.collectors import SyncDataCollector
+from torchrl.collectors.utils import split_trajectories
+from torchrl.data.tensor_specs import CompositeSpec, UnboundedContinuousTensorSpec
+from torchrl.envs import GymWrapper
+from torchrl.envs.libs.gym import default_info_dict_reader
+from torchrl.envs.utils import _terminated_or_truncated
+from torchrl.objectives import ClipPPOLoss
+from torchrl.objectives.utils import _reduce
+from torchrl.record.loggers.wandb import WandbLogger
+from torchrl.trainers import Trainer
+from torchrl.trainers.trainers import LOGGER_METHODS
+
+from omni.isaac.lab.envs import ManagerBasedRLEnv
+
+
+class TorchRLEnvWrapper(GymWrapper):
+    def __init__(self, env: ManagerBasedRLEnv, categorical_action_encoding=False, **kwargs):
+        super().__init__(env, categorical_action_encoding=categorical_action_encoding, **kwargs)
+        self._ep_len_buf = torch.zeros(env.unwrapped.num_envs, device=env.unwrapped.device)
+        self._ep_reward_buf = torch.zeros(env.unwrapped.num_envs, device=env.unwrapped.device)
+        self._curr_ep_len = torch.zeros(env.unwrapped.num_envs, device=env.unwrapped.device)
+        self._curr_reward_sum = torch.zeros(env.unwrapped.num_envs, device=env.unwrapped.device)
+        self._env_reset_mask = torch.zeros(env.unwrapped.num_envs, device=env.unwrapped.device, dtype=torch.bool)
+
+    def maybe_reset(self, tensordict: TensorDictBase) -> TensorDictBase:
+        """Checks the done keys of the input tensordict. Unlike the base GymWrapper implementation, we do not
+        call env.reset()
```

Comment:
interesting, we bumped into a similar problem with @luisenp and @teopir

We need a better handling of auto-resetting envs in torchrl.

How do you get the last values of a trajectory (if you can get them)?